adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -152,8 +167,27 @@ private void validateAcquireTokenRequest(final AuthenticationRequest authenticat
         }
 
         // validate authority
+        Telemetry.getInstance().startEvent(authenticationRequest.getTelemetryRequestId(),
```

Comment:
That's strange. Why startEvent doesn't get an event itself or returns an event? Why would we even need to call `start`? Can we instead just save everything we need in the event and only have 1 function call Telemetry.getInstance().logEvent()?